### PR TITLE
Added clean cli interface for linked list with added functions

### PR DESCRIPTION
**Pull request by Arunkumar0258**
**Fixes issue:**

Issue number: 3994

**Changes:**

Added 2 files in which C file invokes cli based interface functions and linked list operations. The header file is where all the functions are implemented.

### CODE REVIEW

File: code/data_structures/src/list/singly_linked_list/menu_interface/link.h [hunk 1]
```diff
@@ -0,0 +1,397 @@
+#include <stdio.h>
```
Comment (arnavb) [inline code/data_structures/src/list/singly_linked_list/menu_interface/link.h]: Please follow our C style guide (you can find it from the root directory/guides/c/README.md)

File: code/data_structures/src/list/singly_linked_list/menu_interface/singly_list.c [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+					break;
+			case 16:
+					listfile();
+					printf("Enter the name of file to read from: ");
+					scanf("%s",filename);
+					readfile(filename);
+					break;
+			case 17:
+					return n;
+					break;
```
Comment (arnavb) [inline code/data_structures/src/list/singly_linked_list/menu_interface/singly_list.c]: Break is useless here.

File: code/data_structures/src/list/singly_linked_list/menu_interface/singly_list.c [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+	printf("13. Reverse\n");
+	printf("14. Sort(Ascending)\n");
+	printf("15. Write to File\n");
+	printf("16. Read from a file\n");
+	printf("17. Quit\n");
+}
+
+int menu()
+{
+	int n,a,b;
```
Comment (arnavb) [inline code/data_structures/src/list/singly_linked_list/menu_interface/singly_list.c]: Please declare these variables closer to where they are used.

File: code/data_structures/src/list/singly_linked_list/menu_interface/singly_list.c [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+	char *filename = malloc(MAX_LENGTH*sizeof(char));
+	clear();
+	usleep(500000);
```
Comment (arnavb) [inline code/data_structures/src/list/singly_linked_list/menu_interface/singly_list.c]: Why?

File: code/data_structures/src/list/singly_linked_list/menu_interface/singly_list.c [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+					break;
+			case 17:
+					return n;
+					break;
+		}
+}
+
+void clear(void)
+{
+	printf("\033[2J");
```
Comment (arnavb) [inline code/data_structures/src/list/singly_linked_list/menu_interface/singly_list.c]: Is this 100% cross-platform?

File: code/data_structures/src/list/singly_linked_list/menu_interface/link.h (hunk 1 above)
Comment (Arunkumar0258) [reply code/data_structures/src/list/singly_linked_list/menu_interface/link.h]: I have changed the code in both files according to C coding style mentioned in above file. Kindly see it.

File: code/data_structures/src/list/singly_linked_list/menu_interface/singly_list.c (hunk 2 above)
Comment (Arunkumar0258) [reply code/data_structures/src/list/singly_linked_list/menu_interface/singly_list.c]: Break statement has been removed. Thanks for the suggestion.

File: code/data_structures/src/list/singly_linked_list/menu_interface/singly_list.c (hunk 3 above)
Comment (Arunkumar0258) [reply code/data_structures/src/list/singly_linked_list/menu_interface/singly_list.c]: These variables are used in that function only hence I think they were positioned right. But I have moved the declaration slightly below.

File: code/data_structures/src/list/singly_linked_list/menu_interface/singly_list.c (hunk 4 above)
Comment (Arunkumar0258) [reply code/data_structures/src/list/singly_linked_list/menu_interface/singly_list.c]: It is used for smooth transition in UI. It's for 500ms only. If no delay is given, interface can be messy. Also clear is used to clear screen after any operation is done. File is used for writing linked list in a file and reading a linked list from a given file.

File: code/data_structures/src/list/singly_linked_list/menu_interface/singly_list.c (hunk 5 above)
Comment (Arunkumar0258) [reply code/data_structures/src/list/singly_linked_list/menu_interface/singly_list.c]: Thanks for the suggestion. I have made the clear() function cross-platform in Windows, Linux and MacOS.

**Comment by Arunkumar0258** [comment]
@arnavb Can you review the changes I made and address them?

File: code/data_structures/src/list/singly_linked_list/menu_interface/link.h [hunk 6]
```diff
@@ -0,0 +1,411 @@
...
+	int info;
+	struct list* next;
+}list;
+
+list *start=NULL,*node=NULL,*curr=NULL,*prev=NULL;
+
+void insert_beg(int);
+void insert_end(int);
+void insert_at_loc(int,int);
+void insert_after_value(int,int);
```
Comment (arnavb) [inline code/data_structures/src/list/singly_linked_list/menu_interface/link.h]: You need spaces after all commas.

Comment (Arunkumar0258) [reply code/data_structures/src/list/singly_linked_list/menu_interface/link.h]: I have added commas.

**Comment by Arunkumar0258** [comment]
@arnavb I have added the commas. Please review.

**Comment by Arunkumar0258** [comment]
@arnavb Can you please review!

**Comment by arnavb** [comment]
djldk

**Pull request merged**